### ISSUE TITLE
[V3] Implement new Concept Exercise: comparisons

### ISSUE THREAD

**Issue by DavidGerva** [exercism/python#2288]
This issue describes how to implement the `comparisons` concept exercise for the Python track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

This concept exercise should teach how basic (_non-customized_) comparisons work in python and how to use them effectively.

## Learning objectives

- understand all [comparison operations](https://docs.python.org/3/reference/expressions.html#comparisons) in Python have the same _priority_ and are evaluated _after_ arithmetic, shifting, or bitwise operations.
- understand all comparisons yield the `boolean` values `True` and `False`
- know that _identity_ comparisons `is` and `is not` are for checking an objects _identity_ only
- understand that `==` and `!=` compare both the _value_ & _type_ of an object.
- know where Python has altered the behavior of `==` and `!=` for certain `built-in` types (such as [numbers](https://docs.python.org/3/library/stdtypes.html#typesnumeric)), or for standard library types like [decimals](https://docs.python.org/3/library/decimal.html#decimal.Decimal), and [fractions](https://docs.python.org/3/library/fractions.html#fractions.Fraction) to allow comparison _across_ and _within_ type.
- know that unlike numeric types, strings (`str`) and binary sequences ([`bytes`](https://docs.python.org/3/library/stdtypes.html#bytes) & [`byte array`](https://docs.python.org/3/library/stdtypes.html#bytearray)) **cannot** be directly compared.
- understand how comparisons work  within `built-in` [sequence types](https://docs.python.org/3/library/stdtypes.html#sequence-types-list-tuple-range) (`list`, `tuple`, `range`) and `built-in` collection types ([`set`](https://docs.python.org/3/library/stdtypes.html#set-types-set-frozenset), [`dict`](https://docs.python.org/3/library/stdtypes.html#mapping-types-dict))
- know about the "special" comparisons `None`, `NotImplemented` (_comparing either should use identity operators and not equality operators because they are singleton objects_) and `NaN` (_NaN is **never** `==` to itself_)
- use the _value comparison operators_ `==`, `>`, `<`, `!=` with _numeric types_
- use the _value comparison operators_ `==`, `>`, `<`, `!=` with _non-numeric types_
- use `is` and `is not` to check/verify identity

## Out of scope

- rich comparison with `__lt__`, `__le__`, `__ne__`, `__ge__`, `__gt__`
- understanding (_and using the concept_) that the `==` operator calls the dunder method `__eq__()` on a specific object, and uses that object's implementation for comparison.  Where no implementation is present, the default `__eq__()` from _generic_ `object` is used.
- overloading the default implementation of the `__eq__()` dunder method on a specific object to customize comparison behavior.
- `set operations`
- performance considerations

## Concepts

- Comparison priority in Python
- Comparison operators`==`, `>`, `<`, `!=`
- Identity methods `is` and `is not`
- Equality applied to `built-in` types
- Equivalence vs equality
- Inequality

## Prerequisites

- `basics`
- `bools`
- `dicts`
- `lists`
- `sets`
- `strings`
- `tuples`
- `numbers`
- `iteration`

## Resources to refer to

- [Comparisons in Python (Python language reference)](https://docs.python.org/3/reference/expressions.html#comparisons)
- [Value comparisons in Python (Python language reference)](https://docs.python.org/3/reference/expressions.html#value-comparisons)
- [Identity comparisons in Python (Python language reference)](https://docs.python.org/3/reference/expressions.html#is-not)
- [Python operators official doc](https://docs.python.org/3/library/operator.html)
- [Python Object Model (Python docs)](https://docs.python.org/3/reference/datamodel.html#objects)
- [Basic Customization](https://docs.python.org/3/reference/datamodel.html#customization)
- [Python basic operators on tutorialspoint](https://www.tutorialspoint.com/python/python_basic_operators.htm)
- [Python comparison operators on data-flair](https://data-flair.training/blogs/python-comparison-operators/)
- [PEP 207 to allow Operator Overloading for Comparison](https://www.python.org/dev/peps/pep-0207/)

### Hints

- Referring to one or more of the resources linked above, or analogous resources from a trusted source.

## Concept Description 
(_a variant of this can be used for the `v3/languages/python/concepts/<concept>/about.md` doc and this exercises  `introduction.md` doc._)

_**The concept description needs to be filled in here.**_

## Representer

No changes required.

## Analyzer

No changes required.

## Implementing

Tests should be written using unittest.TestCase and the test file named comparisons_test.py.

Code in the `.meta/example.py` file should only use syntax & concepts introduced in this exercise or one of its prerequisites. 
Please do not use comprehensions, generator expressions, or other syntax not previously covered.  Please also follow [PEP8](https://www.python.org/dev/peps/pep-0008/) guidelines.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by BethanyG** [comment]
Will be closing this issue, since [PR #2420](https://github.com/exercism/python/pull/2420) completes this.  Any edits/additions/re-writes will be noted under a different issue.
